core/src/main/java/org/apache/iceberg/OverwriteData.java

```diff
@@ -88,6 +121,40 @@ public OverwriteFiles validateAddedFiles() {
       }
     }
 
+    if (conflictingAppendsRowFilter != null) {
+      PartitionSpec spec = writeSpec();
+      Expression inclusiveExpr = Projections.inclusive(spec).project(conflictingAppendsRowFilter);
+      Evaluator inclusive = new Evaluator(spec.partitionType(), inclusiveExpr);
+
+      List<DataFile> newFiles = collectNewFiles(base);
+      for (DataFile newFile : newFiles) {
+        // we do partition-level conflict resolution right now
+        // we can enhance it by leveraging column stats and MetricsEvaluator
```

Comment:
Is there a reason not to add these optimizations now? If you don't want to add these, then let's at least add a follow-up issue for it.